ocaml/xapi/xapi_gpumon.ml

```diff
@@ -77,10 +77,11 @@ module Nvidia = struct
       let dbg   = Context.string_of_task __context in
       let vm    = Db.VGPU.get_VM ~__context ~self:vgpu in
       let domid = Db.VM.get_domid ~__context ~self:vm |> Int64.to_int in
+      let vgpu_uuid = Db.VGPU.get_uuid ~__context ~self:vgpu in
       Db.VGPU.get_resident_on ~__context ~self:vgpu
       |> (fun self -> Db.PGPU.get_PCI ~__context ~self)
       |> (fun self -> Db.PCI.get_pci_id ~__context ~self)
-      |> Gpumon_client.Client.Nvidia.get_vgpu_metadata dbg domid
+      |> (fun pgpu_address -> Gpumon_client.Client.Nvidia.get_vgpu_metadata dbg domid pgpu_address vgpu_uuid)
       |> (function
           | []      -> []
           | [meta]  -> [key, Stdext.Base64.encode meta]
```

Comment:
I think the error message below should report the UUID. What happens is that we found more than one vGPU for a given domid and uuid.

Reply:
Updated.